### ISSUE TITLE
New check for "grep | head -1" which should inform that "grep -m1" is better

### ISSUE THREAD

**Issue by thomasmerz** [koalaman/shellcheck#2505]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
#!/bin/bash
cmd="grep \"javaExe=\" $sJavaNew | head -1"
```

#### Here's what shellcheck currently says:
It says: nothing.
But it should tell users that there's a "-m NUM, --max-count=NUM" parameter which makes `grep` "Stop reading a file after NUM matching lines."

#### Here's what I wanted or expected to see:
Therefore the example above should be corrected to this:
```
cmd="grep -m1 \"javaExe=\" $sJavaNew"
```

**Comment by koalaman** [comment]
Thanks for the suggestion! It can definitely look cleaner when supported, but `-m` is not portable so it's not unambiguously better.
